# Step 3A. Alignment-free quantification using Salmon

This step builds a Salmon index and quantifies paired-end reads.  
**Note:** Copy & paste the commands into your terminal and replace placeholders `<>`.

## Commands

```bash
# Build Salmon index
salmon index \
  -t <transcripts.fa>   # reference transcriptome FASTA
  -i <index_name>       # Salmon index directory name/path
  -p <threads>          # number of threads for indexing

# Quantification (per sample)
salmon quant \
  -i <index_name>       # Salmon index directory
  -l A                  # auto-detect library type
  -1 <read1.fastq.gz>   # paired-end read1 FASTQ
  -2 <read2.fastq.gz>   # paired-end read2 FASTQ
  -p <threads>          # number of threads for quant
  --validateMappings    # enable selective alignment
  -o <output_dir>       # output dir (contains quant.sf)
```

## Example

```
# Build index
salmon index \
  -t ref/transcripts.fa \
  -i ref/salmon_index \
  -p 8

# Quantify one sample
salmon quant \
  -i ref/salmon_index \
  -l A \
  -1 data/S1_R1.fastq.gz \
  -2 data/S1_R2.fastq.gz \
  -p 8 \
  --validateMappings \
  -o results/salmon/S1
```
### Code Explanation:
· **samples.tsv**: a tab-delimited file like:
```
sample_id condition salmon_dir
S1 A /path/to/S1
S2 A /path/to/S2
S3 B /path/to/S3
```
At minimum, it should include `sample_id`, `salmon_dir`.

· **quant.sf**: Output file generated by Salmon for each sample, containing transcript-level abundance estimates (TPM, NumReads, EffectiveLength).  
These files serve as the input for gene-level summarization.  

· **tx2gene.tsv**: a tab-delimited file like:
```
transcript_id gene_id
ENST00000456328 ENSG00000223972
ENST00000450305 ENSG00000223972
ENST00000515242 ENSG00000227232
```
At minimum, it should include `transcript_id` and `gene_id`.  

**Note**: Use the same reference annotation (version) as used to build the Salmon index to ensure consistent ID mapping. 
